filter_spatial.json

```diff
@@ -0,0 +1,40 @@
+{
+    "id": "filter_spatial",
+    "summary": "Spatial filter using geometries",
+    "description": "Limits the data cube over the spatial dimensions to the specified geometries.\n\n- For **polygons**, the filter retains a pixel in the data cube if the point at the pixel center intersects with at least one of the polygons (as defined in the Simple Features standard by the OGC).\n- For **points** and **lines** (line strings), the filter retains all the pixel centers closest to any of the geometries.\n\nMore specifically, pixels outside of the bounding box of the given geometry will not be available after filtering. All pixels inside the bounding box that are not retained will be set to `null` (no data).",
+    "categories": [
+        "filter"
+    ],
+    "parameters": [
+        {
+            "name": "data",
+            "description": "A data cube.",
+            "schema": {
+                "type": "object",
+                "subtype": "raster-cube"
+            }
+        },
+        {
+            "name": "geometries",
+            "description": "One or more geometries used for filtering, specified as GeoJSON.",
+            "schema": {
+                "type": "object",
+                "subtype": "geojson"
+            }
+        }
+    ],
+    "returns": {
+        "description": "A data cube restricted to the specified geometries. Therefore, the cardinality is potentially lower, but the resolution and the number of dimensions are the same as for the original data cube.",
```

Comment:
This should be a list of data cubes. If we have several polygons, then the result should be a data cube per polygon. Otherwise, you would need to change the above to "a single geometry" and allow for some kind of iterator, though that would be quite complex on the process graph...

Reply:
Why can't it just be a single data cube with all things outside of the geometries set to null?

Reply:
Then it will be very sparse. If you use two points, one in Africa and one in North America, you don't want a raster of almost entirely NAs, because you are not interested in those.

Reply:
Indeed, so two separate calls to the process would be better. In most cases I hope users will have two spatially restricted data cubes anyway to avoid loading huge datacubes at startup.

Reply:
Yes, that's fine (if there is a way to iterate over multiple geometries; I'm not completely sure if there is?), but then is there any reason to support multiple geometries in this process? We can just say that it's only for a single geometry. If they do want a sparse raster, they can make a union of the geometry.

Reply:
Well, I guess there are valid use cases, e.g. passing the geometries of country borders in Europe and getting zonal statistics for them for a data cube with data of Europe. It really depends on your use case. If you want a pixel in New York and a Pixel in Dubai you are probably better using two separate data cubes, which you could iterate over with array_apply in theory. Or just use user-defined processes and issue two jobs (that's probably what back-ends support first).

You can't really iterate over the output (= vector cube) of the process at the moment. As I said in another comment, we only allow to save the results at the moment. In theory, it would be a call to apply_dimension in the future.